application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -0,0 +1,348 @@
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.BanSystem;
+import org.togetherjava.tjbot.db.generated.tables.KickSystem;
+import org.togetherjava.tjbot.db.generated.tables.WarnSystem;
+import org.togetherjava.tjbot.db.generated.tables.records.BanSystemRecord;
+import org.togetherjava.tjbot.db.generated.tables.records.KickSystemRecord;
+import org.togetherjava.tjbot.db.generated.tables.records.WarnSystemRecord;
+
+import java.awt.*;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import static org.togetherjava.tjbot.commands.utils.MessageUtils.replyEphemeral;
+
+/**
+ * This command can be used to retrieve data for moderation commands. For example this command
+ * allows you to retrieve how many warns a certain user has.
+ */
+public class AuditCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(AuditCommand.class);
+    private static final String ACTION_VERB = "audit";
+    private static final String WARN_SUBCOMMAND = "warn";
+    private static final String WARN_USER_OPTION = "user";
+    private static final String WARNED = "warned";
+    private static final String KICK_SUBCOMMAND = "kick";
+    private static final String KICK_USER_OPTION = "user";
+    private static final String KICKED = "kicked";
+    private static final String BAN_SUBCOMMAND = "ban";
+    private static final String BAN_USER_OPTION = "user";
+    private static final String BANNED = "banned";
+    private static final String DESCRIPTION = "The user who you want to get the values for";
+    private static final String OPTION_IS_NULL = "The option is null";
+    private static final String GUILD_IS_NULL = "he guild is null";
+    private static final String THE_USER = "the user ";
+    private static final String COLOR_CODE = "#895FE8";
+    private final Database database;
+    private final Predicate<String> hasKickANDWarnRequiredRole;
+    private final Predicate<String> hasBanRequiredRole;
+
+
+    /**
+     * Creates a new adapter with the given data.
+     *
+     * @param database used to retrieve the user data for the moderation commands
+     */
+    public AuditCommand(Database database) {
+        super(ACTION_VERB, "get the audit for the commands", SlashCommandVisibility.GUILD);
+        this.database = database;
+
+        getData().addSubcommands(
+                new SubcommandData(WARN_SUBCOMMAND, "Gets the warn values for that user")
+                    .addOption(OptionType.USER, WARN_USER_OPTION, DESCRIPTION, true),
+                new SubcommandData(KICK_SUBCOMMAND, "Gets the kick values for that user")
+                    .addOption(OptionType.USER, KICK_USER_OPTION, DESCRIPTION, true),
+                new SubcommandData(BAN_SUBCOMMAND, "Gets the ban values for that user")
+                    .addOption(OptionType.USER, BAN_USER_OPTION, DESCRIPTION, true));
```

Comment:
The design on this command is questionable. Change it to return all logged and known moderation actions against the user, in chronological order.

So `/audit user` and that returns a full list of everything for that user in order.